frontend/src/app/features/details/parcelDescriptions/parcelDescriptions.tsx

```diff
@@ -1,59 +1,74 @@
 import SearchInput from '../../../components/search/SearchInput';
 import Sort from '../../../components/sort/Sort';
-import Table from '../../../components/table/Table';
-import { RequestStatus } from '../../../helpers/requests/status';
 import { TableColumn } from '../../../components/table/TableColumn';
 import { useEffect } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
 import { AppDispatch, RootState } from '../../../Store';
 import {
   fetchParcelDescriptions,
-  IFetchParcelDescriptionParams,
-  IParcelDescriptionState,
-  setCurrentPage,
-  setResultsPerPage,
-  setSearchParam,
-  setSortBy,
-  setSortByDir,
-  setSortByInputValue,
+  initialParcelDescriptionsState,
 } from './parcelDescriptionsSlice';
 import { columns } from './parcelDescriptionsConfig';
 import { useParams } from 'react-router-dom';
 import ParcelDescriptionTable from './ParcelDescriptionTable';
+import { IParcelDescriptionsState } from './parcelDescriptionsInterfaces';
+import { RequestStatus } from '../../../helpers/requests/status';
 
 const ParcelDescriptions = () => {
   const dispatch = useDispatch<AppDispatch>();
-  const parcelDescriptionsState: IParcelDescriptionState = useSelector(
-    (state: RootState) => state.parcelDescriptions,
-  );
+  let {
+    siteId,
+    data,
+    requestStatus,
+    totalResults,
+    currentPage,
+    resultsPerPage,
+    searchParam,
+    sortBy,
+    sortByDir,
+    sortByInputValue,
+    needsUpdate,
```

Comment:
Every time the Redux state updates, useSelector will re-run its selector function. So here as you are selecting the root state , IMO every time any state changes in application will cause parcel description component to re-render, which is not ideal and lead to performance and other issues. I would recommend to move this code to parcel description slice and expose Individual selectors as needed. (Similar way we implemented in other components)